pyshelf/search_portal.py

```diff
@@ -1,20 +1,86 @@
-from pyshelf.search.type import Type as SearchType
-from pyshelf.search.metadata import Metadata
+from pyshelf.metadata.keys import Keys as MetadataKeys
+from pyshelf.resource_identity import ResourceIdentity
 
 
 class SearchPortal(object):
+    """
+        This class is the link between the request/view and the search layer.
+    """
     def __init__(self, container):
         self.container = container
-        self.search_manager = self.container.search.search_manager
+        self.search_manager = self.container.search.manager
+        self.search_parser = self.container.search_parser
+        self.link_manager = self.container.link_manager
+        self.resource_id = self.container.resource_identity
 
     def search(self, criteria):
         """
-            Searches ElasticSearch metadata documents.
+            Searches based on criteria defined in request and assigns links to response
+            for each search hit.
 
             Args:
-                criteria(dict): Criteria to use to initiate search.
+                criteria(dict): Search and sort criteria formatted as show below.
 
             Format of criteria:
+                {
+                    "search": [
+                        "version~=1.1"
+                    ],
+                    "sort": [
+                        "version, VERSION, ASC",
+                        "bob,DESC"
+                    ],
+                    "limit": 1
+                }
+
+                Or if there is only a single sort/search
+
+                {
+                    "search": "version~=1.1",
+                    "sort": "version, VERSION, ASC",
+                    "limit": 1
+                }
+        """
+        criteria["search"] = self._default_to_list(criteria.get("search"))
+        criteria["sort"] = self._default_to_list(criteria.get("sort"))
+        path_search = "{0}={1}*".format(MetadataKeys.PATH, self.resource_id.resource_path)
+        criteria["search"].append(path_search)
+
+        formatted_criteria = self.search_parser.from_request(criteria)
+        results = self.search_manager.search(formatted_criteria)
+        artifact_list = self._list_artifacts(results, criteria.get("limit"))
+        self.link_manager.assign_listing(artifact_list)
+
+    def _default_to_list(self, criteria):
```

Comment:
Lol so I moved the path format out because of your original code review comments which required the defaulting to a list to be in the portal. I am going to go ahead and pass the formatted path in separately so that can be in the portal but list stuff can be in the parser.